### ISSUE TITLE
Model Compilation Fails with Recent Transformers (FileNotFound pytorch_model.bin)

### ISSUE THREAD

**Issue by drocco007** [huggingface/optimum-neuron#399]
When following the [llama2-13b-chatbot](https://huggingface.co/docs/optimum-neuron/tutorials/llama2-13b-chatbot) tutorial, the export step in section 1 fails with recent versions of `transformers`:

```
(torch_newest) sh-5.2$ pip list | egrep '(torch|transformers|optimum|neuron)'
aws-neuronx-runtime-discovery 2.9
libneuronxla                  0.5.538
neuronx-cc                    2.11.0.34+c5231f848
neuronx-distributed           0.5.0
neuronx-hwm                   2.11.0.2+e34678757
optimum                       1.16.1
optimum-neuron                0.0.13
torch                         1.13.1
torch-neuronx                 1.13.1.1.12.0
torch-xla                     1.13.1+torchneuronc
torchvision                   0.14.1
transformers                  4.36.2
transformers-neuronx          0.8.268

(torch_newest) sh-5.2$ python
Python 3.9.16 (main, Sep  8 2023, 00:00:00)
[GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from optimum.neuron import NeuronModelForCausalLM
>>> compiler_args = {"num_cores": 12, "auto_cast_type": 'fp16'}
>>> input_shapes = {"batch_size": 1, "sequence_length": 2048}
>>> model = NeuronModelForCausalLM.from_pretrained(
...         "NousResearch/Llama-2-13b-chat-hf",
...         export=True,
...         **compiler_args,
...         **input_shapes)
config.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 608/608 [00:00<00:00, 176kB/s]
model.safetensors.index.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 33.4k/33.4k [00:00<00:00, 12.8MB/s]
model-00001-of-00003.safetensors: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9.95G/9.95G [00:21<00:00, 469MB/s]
model-00002-of-00003.safetensors: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9.90G/9.90G [00:21<00:00, 463MB/s]
model-00003-of-00003.safetensors: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6.18G/6.18G [00:13<00:00, 453MB/s]
Downloading shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:56<00:00, 18.81s/it]
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:01<00:00,  1.67it/s]
generation_config.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 175/175 [00:00<00:00, 65.9kB/s]
/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers/generation/configuration_utils.py:389: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers/generation/configuration_utils.py:394: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `0.6` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers/generation/configuration_utils.py:557: UserWarning: The generation config instance is invalid -- `.validate()` throws warnings and/or exceptions. Fix these issues to save the configuration. This warning will be raised to an exception in v4.34.

Thrown during validation:
`do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`.
  warnings.warn(
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/optimum/modeling_base.py", line 372, in from_pretrained
    return from_pretrained_method(
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/optimum/neuron/modeling_decoder.py", line 153, in _from_transformers
    return cls._from_pretrained(checkpoint_dir, config)
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/optimum/neuron/modeling_decoder.py", line 209, in _from_pretrained
    neuronx_model = exporter.neuronx_class.from_pretrained(
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers_neuronx/module.py", line 148, in from_pretrained
    state_dict = torch.load(state_dict_path)
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/torch/serialization.py", line 771, in load
    with _open_file_like(f, 'rb') as opened_file:
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/torch/serialization.py", line 270, in _open_file_like
    return _open_file(name_or_buffer, mode)
  File "/home/ssm-user/torch_newest/lib64/python3.9/site-packages/torch/serialization.py", line 251, in __init__
    super(_open_file, self).__init__(open(name, mode))
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpi46jztoc/pytorch_model.bin'

```

Downgrading `transformers` allows the example to succeed:

```
# transformers 4.33.1 is the lowest version supported by optimum-neuron 0.0.13
(torch_newest) $ pip install 'transformers==4.33.1'
...
(Python 3.9.16 (main, Sep  8 2023, 00:00:00)
[GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from optimum.neuron import NeuronModelForCausalLM
>>> compiler_args = {"num_cores": 12, "auto_cast_type": 'fp16'}
>>> input_shapes = {"batch_size": 1, "sequence_length": 2048}
>>> model = NeuronModelForCausalLM.from_pretrained(
...         "NousResearch/Llama-2-13b-chat-hf",
...         export=True,
...         **compiler_args,
...         **input_shapes)
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:02<00:00,  1.24it/s]
/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers/generation/configuration_utils.py:362: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers/generation/configuration_utils.py:367: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `0.6` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/home/ssm-user/torch_newest/lib64/python3.9/site-packages/transformers/generation/configuration_utils.py:508: UserWarning: The generation config instance is invalid -- `.validate()` throws warnings and/or exceptions. Fix these issues to save the configuration. This warning will be raised to an exception in v4.34.

Thrown during validation:
`do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`.
  warnings.warn(
2024-01-08 21:21:09.000548:  52865  INFO ||NEURON_CACHE||: Compile cache path: /var/tmp/neuron-compile-cache
2024-01-08 21:21:09.000550:  52865  INFO ||NEURON_CC_WRAPPER||: Call compiler with cmd: ['neuronx-cc', '--target=trn1', 'compile', '--framework', 'XLA', '/tmp/no-user/neuroncc_compile_workdir/72d38c26-9c17-45f1-8fdd-2e24ea9a1c81/model.MODULE_021264b269c7c788ab16+8737852b.hlo.pb', '--output', '/tmp/no-user/neuroncc_compile_workdir/72d38c26-9c17-45f1-8fdd-2e24ea9a1c81/model.MODULE_021264b269c7c788ab16+8737852b.neff', '--model-type=transformer', '--model-type=transformer', '--verbose=35']
2024-01-08 21:21:09.000593:  52866  INFO ||NEURON_CACHE||: Compile cache path: /var/tmp/neuron-compile-cache
2024-01-08 21:21:09.000595:  52866  INFO ||NEURON_CC_WRAPPER||: Call compiler with cmd: ['neuronx-cc', '--target=trn1', 'compile', '--framework', 'XLA', '/tmp/no-user/neuroncc_compile_workdir/f6e0f080-3741-41f4-84b2-76eac80dcc7a/model.MODULE_72b1d14dd28a43b41348+8737852b.hlo.pb', '--output', '/tmp/no-user/neuroncc_compile_workdir/f6e0f080-3741-41f4-84b2-76eac80dcc7a/model.MODULE_72b1d14dd28a43b41348+8737852b.neff', '--model-type=transformer', '--model-type=transformer', '--verbose=35']
2024-01-08 21:21:09.000640:  52867  INFO ||NEURON_CACHE||: Compile cache path: /var/tmp/neuron-compile-cache
2024-01-08 21:21:09.000642:  52867  INFO ||NEURON_CC_WRAPPER||: Call compiler with cmd: ['neuronx-cc', '--target=trn1', 'compile', '--framework', 'XLA', '/tmp/no-user/neuroncc_compile_workdir/9733945e-e1ac-44bc-ba27-37a31373cfee/model.MODULE_82b65908029a58094b99+8737852b.hlo.pb', '--output', '/tmp/no-user/neuroncc_compile_workdir/9733945e-e1ac-44bc-ba27-37a31373cfee/model.MODULE_82b65908029a58094b99+8737852b.neff', '--model-type=transformer', '--model-type=transformer', '--verbose=35']
2024-01-08 21:21:09.000685:  52868  INFO ||NEURON_CACHE||: Compile cache path: /var/tmp/neuron-compile-cache
2024-01-08 21:21:09.000687:  52868  INFO ||NEURON_CC_WRAPPER||: Call compiler with cmd: ['neuronx-cc', '--target=trn1', 'compile', '--framework', 'XLA', '/tmp/no-user/neuroncc_compile_workdir/b0631142-6382-47dd-9b68-7b756cb43b56/model.MODULE_2f9209f10ce1c42bd256+8737852b.hlo.pb', '--output', '/tmp/no-user/neuroncc_compile_workdir/b0631142-6382-47dd-9b68-7b756cb43b56/model.MODULE_2f9209f10ce1c42bd256+8737852b.neff', '--model-type=transformer', '--model-type=transformer', '--verbose=35']
2024-01-08 21:21:09.000730:  52869  INFO ||NEURON_CACHE||: Compile cache path: /var/tmp/neuron-compile-cache
2024-01-08 21:21:09.000732:  52869  INFO ||NEURON_CC_WRAPPER||: Call compiler with cmd: ['neuronx-cc', '--target=trn1', 'compile', '--framework', 'XLA', '/tmp/no-user/neuroncc_compile_workdir/6fd02e94-a31f-45bb-bf20-686d899b3f27/model.MODULE_248ec8861a3c79d7dbee+8737852b.hlo.pb', '--output', '/tmp/no-user/neuroncc_compile_workdir/6fd02e94-a31f-45bb-bf20-686d899b3f27/model.MODULE_248ec8861a3c79d7dbee+8737852b.neff', '--model-type=transformer', '--model-type=transformer', '--verbose=35']
....................
```

Instance: inf2.24xlarge 
AMI: Amazon Linux 2023

**Comment by dacorvo** [comment]
Thank you for reporting that issue: this has been fixed in version 0.0.14 (https://github.com/huggingface/optimum-neuron/pull/320).

**Comment by michaelbenayoun** [comment]
Closing since it seems to be resolved. Feel free to re-open if you face the issue again.
